Add rake task to collect admin overview metrics

jacobherrington opened the pull request:
<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 Read the Forem Contributing Guide: https://github.com/forem/forem/blob/master/CONTRIBUTING.md#create-a-pull-request.
     - 📖 Read the Forem Code of Conduct: https://github.com/forem/forem/blob/master/CODE_OF_CONDUCT.md.
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.

     NOTE: Pull Requests from forked repositories will need to be reviewed by
     a Forem Team member before any CI builds will run. Once your PR is approved
     with a `/ci` reply to the PR, it will be allowed to run subsequent builds without
     manual approval.
-->

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

A quick rake task to grab Ahoy events tracking which links have been clicked in the admin overview.

Usage: `bundle exec rake metrics:overview`

## Related Tickets & Documents

Closes https://github.com/forem/internalEngineering/issues/400

## QA Instructions, Screenshots, Recordings

Example:

```
$ bundle exec rake metrics:overview
/admin/invitations: 1
/admin/welcome: 1
```

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

- [ ] Yes
- [x] No, and this is why: We probably don't need to be testing a temporary Rake task
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

rhymes commented on lib/tasks/metrics.rake:
```diff
@@ -0,0 +1,6 @@
+namespace :metrics do
+  desc "Collects metrics from Forem instances"
+  task overview: :environment do
+    puts Ahoy::Event.overview_metrics_to_csv
```
I'm not sure if this happens in production as well but when a rake task is loaded, a bunch of warnings are printed on the standard output, and with `rake metrics:overview > filename.csv` those warnings will end up in the CSV file. On my local installation this is what I found in the file:

```csv
Unset ENV variable: IMGPROXY_KEY.
Unset ENV variable: IMGPROXY_SALT.
DEPRECATION WARNING: Initialization autoloaded the constants RateLimitChecker, ForemStatsDriver, ForemStatsDrivers, ActsAsForemStatsDriver, and ForemStatsDrivers::DatadogDriver.

Being able to do this is deprecated. Autoloading during initialization is going
to be an error condition in future versions of Rails.

Reloading does not reboot the application, and therefore code executed during
initialization does not run again. So, if you reload RateLimitChecker, for example,
the expected changes won't be reflected in that stale Class object.

These autoloaded constants have been unloaded.

Please, check the "Autoloading and Reloading Constants" guide for solutions.
 (called from <main> at /forem/config/environment.rb:5)
Unset ENV variable: RELEASE_FOOTPRINT.
** [Honeybadger] Initializing Honeybadger Error Tracker for Ruby. Ship it! version=4.7.3 framework=rails level=1 pid=49260
  [1m[36mProfileField Load (0.8ms)[0m  [1m[34mSELECT "profile_fields".* FROM "profile_fields" ORDER BY "profile_fields"."id" ASC LIMIT $1[0m  [["LIMIT", 1000]]
  ↳ app/models/profile.rb:55:in `refresh_attributes!'
DEPRECATION WARNING: Devise::Models::Authenticatable::BLACKLIST_FOR_SERIALIZATION is deprecated! Use Devise::Models::Authenticatable::UNSAFE_ATTRIBUTES_FOR_SERIALIZATION instead. (called from const_get at /forem/vendor/cache/devise-0cd72a56f984/lib/devise/models.rb:90)
  [1m[36mAhoy::Event Load (0.8ms)[0m  [1m[34mSELECT "ahoy_events".* FROM "ahoy_events" WHERE "ahoy_events"."name" = $1 ORDER BY "ahoy_events"."id" ASC LIMIT $2[0m  [["name", "Overview Link Clicked"], ["LIMIT", 1000]]
  ↳ app/models/ahoy/event.rb:17:in `block in overview_metrics_to_csv'
id,name,user_id,visit_id,time,target
1,Overview Link Clicked,,1,2021-03-17 15:56:51 UTC,https://example.com
```

jacobherrington replied:
Good point, so it probably makes sense to save the data to a file rather than just printing it to the console!

jacobherrington replied:
I think @vaidehijoshi made a solid suggestion - let's not overcomplicate things for now! This should probably just spit something human-readable out for us to copy/paste elsewhere!

rhymes reviewed (commented):
Left a couple of notes :)

jacobherrington commented:
@vaidehijoshi Addressed your comments on this one. I think leaving the scope on the model makes sense (there is no technical reason for this, it just makes sense to my brain), but I can drop that if you feel it shouldn't be there.

Based on your recommendation, I'm not returning the information in CSV format (I thought that might be useful for analysis across Forems due to a uniform format) and instead the output matches your suggestion:
```
$ bundle exec rake metrics:overview
/admin/invitations: 1
/admin/welcome: 1
```

Let me know if we should do anything differently here!

vaidehijoshi reviewed (approved):
This looks like a good starting point! Thanks for addressing all the comments, I'm looking forward to running this across the fleet and digging into the metrics in coming days 👍🏽

rhymes reviewed (approved):
LGTM!

vaidehijoshi commented:
@jacobherrington I know I already approved this, but since it's not merged yet....I thought of one more thing that we should probably include: the Forem instance name! 😅  When systems runs this across the fleet, it would be super helpful to know _which_ Forem each set of stats is associated with :)

`app_domain` would probably work here, methinks

jacobherrington commented:
@vaidehijoshi added!

vaidehijoshi reviewed (approved):
🚀
